### ISSUE TITLE
Tracking Issue for `windows_case_insensitive`

### ISSUE THREAD

**Issue by CDirkx** [rust-lang/rust#86007]
This is a tracking issue for dealing with strings on Windows that are defined to be case-insensitive: environment variable keys, registry keys, file paths, resource handle names etc.
The feature gate for the issue is `windows_case_insensitive`.

### Unresolved Questions

What is the API std should offer to deal with Windows specific case-insensitive comparisons?

**Comment by ChrisDenton** [comment]
I've been thinking about this recently. Filesystem paths are particularly tricky because they handle case insensitivity differently from the rest. On NTFS, case folding is stored in a `$UpCase` file, so it can differ per drive. Also case sensitivity can be turned off [per directory](https://devblogs.microsoft.com/commandline/per-directory-case-sensitivity-and-wsl/#per-directory-case-sensitivity).

For the rest, perhaps as a start, adding a function to `OsStrExt` that uses the OS function to do case insensitive comparisons? Or at least a compare function that somehow allows specifying case behaviour?

**Comment by CDirkx** [comment]
Yeah that was my plan, add `system_cmp` and `system_eq` to `os::windows::OsStrExt` as a start (#86008). Not sure what we should say there about the file system paths though.
